packages/design-system/src/components/NumericInputField/NumericInput.js

```diff
@@ -146,12 +146,14 @@ class NumericInput extends React.PureComponent {
       className
     );
 
+    const inputValue = typeof value === 'string' ? parseInt(value, 10) : value;
```

Comment:
It is still a little bit weak guard because value with type `string` can be not a number string. In that case a `parseInt` function will return `NaN` value. If you then pass a `NaN` value to a input with type number the React will give a warning:
```
Warning: Received NaN for the `value` attribute. If this is expected, cast the value to a string.
```

Reply:
@walaszczykm what do you propose to do? I prefer to remove the string, but don't want to loose backward compatibility :/

Reply:
use the fact that for all non-numeric strings the `parseInt` function will return `NaN` and use it for fallback. The only question is what fallback value should be. A `0` or the empty string `''` which will result in the valid empty input 🤔

Reply:
@dprzybylek  You also are not handling al other types of values like `object`, `array`, `boolean`, `symbol`. I think all of them should be transformed into a default fallback value (maybe empty string).

Additionally, you can add `useMemo` hook as it is a perfect example for it: generating `inputValue` using provided `value` 🙂.

The implementation of those three suggestions could look like this:
```js
  const inputValue = React.useMemo(() => {
    switch (typeof value) {
      case "number":
        return value;

      case "string": {
        const parsed = parseInt(value, 10);
        return isNaN(parsed) ? "" : value;
      }

      default:
        return "";
    }
  }, [value]);
```

in that way, you can handle specific values types and always fall back to default for all other types. That way you have 100% cover for possible variants 🙂

Reply:
But this is only a suggestion. You should think about desired behaviour for wrong values. Maybe relying on default React warning is fine I don't say it is not but it is good to be aware off this.

Reply:
please look at 04f9d89997f4e85af874c527bcfee0f1670183d1